File: binarysearch.cpp
#include <iostream>
using namespace std;
int main()
{
    int arr[100];
    int mid, i, num, search;

    cout << "Enter the size of the array: " << endl;
    cin >> num;
    cout << " Enter the values in sorted array: " << endl;

    for (i = 0; i < num; i++)
    {
        cout << " arr [" << i << "] = ";
        cin >> arr[i];
    }

    int st = 0;
    int end = num - 1;
    cout << "Enter the number to be searched from sorted array: " << endl;
    cin >> search;

    while (st <= end)
    {
        mid = (st + end) / 2;
        if (arr[mid] == search)
        {
            cout << " Element is found at index " << (mid);
            exit(0);
        }
        else if (search > arr[mid])
        {
            st = mid + 1;
        }
        else if (search < arr[mid])
        {
            end = mid - 1;
        }
    }
    cout << "Number is not found. " << endl;
    for (int i = num-1; i > mid; i--)
        arr[i + 1] = arr[i];
      arr[mid+1] = search;
    cout << "Array after insertion:- ";
    for (int i = 0; i < num + 1; i++)
        cout << arr[i] << " ";

    return 0;
}

File: binarySearch.java
public class binarySearch {
    public static void main(String[] args) {
        int[] arr={1,2,3,4,5,6,7,8};
         int target=2;
        System.out.println(binarySearch(arr,target));
    }
    static int  binarySearch(int[] arr,int target){
        int start=0;
        int end=arr.length-1;
        while(start<=end){
            int mid=start-(start-end)/2;
            if   (target<arr[mid]){
                end = mid - 1;
            }
            else if
            ( target>arr[mid]){
                start=mid+1;
            }
            else
                 {
                return mid;
                 }
            }
        return -1;
        }
    }

